packages/core/admin/admin/src/components/AutoReloadOverlayBlocker/index.js

```diff
@@ -26,18 +18,10 @@ const AutoReloadOverlayBlocker = () => {
     setIsOpen(false);
     setState({ start: 0, elapsed: 0 });
     setConfig(undefined);
-
-    if (document.getElementById(ID)) {
-      document.body.removeChild(overlayContainer);
-    }
   };
 
-  useEffect(() => {
-    window.strapi = Object.assign(window.strapi || {}, {
-      lockAppWithAutoreload,
-      unlockAppWithAutoreload,
-    });
-  }, []);
+  const lockApp = useRef(lockAppWithAutoreload);
```

Comment:
useRef is not necessary since it will re-render only when its internal state changes or when StrapiApp will render

Reply:
My concern is more when adding lockApp to a hook dependencies array, using a ref will prevent an infinite loop

Reply:
are we sure it will trigger an infinite loop? I don't understand how it would generate this 🤔